File: 2022.1/estrutura_de_dados_e_algoritmos_II/slides/codigos-operacoes-sobre-arquivos/criaarquivo.c
#include<stdio.h>
#include<stdlib.h>

#include "tiposArquivo.h"

int main() {

	FILE *f;
	
	printf ("Criando arquivo \"dados\" ...\n");
	if ((f = fopen("dados","w+"))) 
		printf ("Arquivo \"dados\" criado com sucesso.\n");
	else {
		printf ("Erro na criacao do arquivo \"dados\".");
		exit(-1);
	}
	
	printf ("Inicializando o arquivo ...\n");

	Registro r;
	r.ocupado = false;

	int i;
	for (i=0; i < MTAMARQUIVO; i++) {
		fwrite (&r, sizeof(Registro), 1, f);
	}
			
	fclose(f);
}
	  

File: 2022.1/estrutura_de_dados_e_algoritmos_II/slides/python_codigos_operacoes/criaArquivo.py
import tipoArquivo
import pickle

print("Criando arquivo \"dados\" ...\n")

f = open("dados", "wb")

if f:
    print("Arquivo \"dados\" criado com sucesso.\n")
else:
    print("Erro na criacao do arquivo \"dados\".")
    exit(-1)

print("Inicializando o arquivo ...\n")

r = tipoArquivo.Registro()
r.ocupado = False

for i in range(tipoArquivo.MTAMARQUIVO):
    f.write(pickle.dumps(r))

f.close()